java-connectors/kafka-connect-gcp-common/src/main/java/io/lenses/streamreactor/connect/gcp/common/config/AuthModeSettings.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.gcp.common.config;
+
+import io.lenses.streamreactor.common.config.base.ConfigMap;
+import io.lenses.streamreactor.connect.gcp.common.auth.mode.*;
+import lombok.AllArgsConstructor;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigException;
+
+@AllArgsConstructor
+public class AuthModeSettings {
+
+    private final String connectorPrefix;
+
+    public String getAuthModeKey() {
+        return generateKey("gcp.auth.mode");
+    }
+
+    public String getCredentialsKey() {
+        return generateKey("gcp.credentials");
+    }
+
+    public String getFileKey() {
+        return generateKey("gcp.file");
+    }
+
+    private String generateKey(String suffix) {
```

Comment:
Thank you for the comment, there's a lot to cover.

I've experimented with this, and thanks to your comment thought about it a bit more so that I can put down my reasons.

`ConnectorPrefixed` is an interface and has a single method, getConnectorPrefix().  You mentioned I should reuse it, but from what I can see it is unused code, so `reusing` it is not really needed - it needs to be used! 😁 

Coming from the old scala implementation - for simplicity's sake, I would like to resist so much inheritance this time around.  Let's [prefer composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance).  Personally I find the scala code difficult to reason about in the area of configuration and I don't want to follow the same patterns if they are not helpful.  In the scala code we mix in a trait to the configuration so we don't have to define the connector prefix in every class, it is mixed in to the actual connector configuration class.

But... I think it's ok to pass the connector prefix around.  The wiring will be minimal and it's a unit easily mocked or constructed for testing.

You mentioned the "old way" (which ironically, is used by the newest connector 🙃 ), in which you define the connector configuration keys as static variables, **that is totally a valid approach** **_until_** you share code between multiple connectors, but then the connector prefix may be different for each connector.  As this one will.  So the code needs to be laid out differently to accommodate this at runtime.

So I'm going to make a few changes to make this clearer:

* Moving the connectorPrefix and generateKey to a separate class (ConnectorPrefix) in commons which we can reuse and pass around.
* Renaming the generateKey method, because you are correct it is confusingly named.
* Setting the authModeKey/credentialsKey/fileKey in the constructor and no longer keeping the connectorPrefix in a class field.

I would also like to get @stheppi 's view on this as it impacts all of us and makes sense to discuss how we do things like this as a team.